packages/edit-site/src/components/home-view-preview/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+
+import Editor from '../editor';
+
+export function HomeViewPreview() {
+	const { isBlockBasedTheme, siteUrl } = useSelect( ( select ) => {
+		const { getEntityRecord, getCurrentTheme } = select( coreStore );
+		const siteData = getEntityRecord( 'root', '__unstableBase' );
+
+		return {
+			isBlockBasedTheme: getCurrentTheme()?.is_block_theme,
+			siteUrl: siteData?.home,
+		};
+	}, [] );
+
+	// If theme is block based, return the Editor, otherwise return the site preview.
+	return isBlockBasedTheme ? (
+		<Editor />
+	) : (
+		<iframe
+			src={ siteUrl }
+			title="front-end view"
+			style={ {
+				display: 'block',
+				width: '100%',
+				height: '100%',
```

Comment:
```suggestion
				height: '100%',
				backgroundColor: 'white',
```

[Just like block themes](https://github.com/WordPress/gutenberg/blob/a5387c179570981d4d48c38ead43ab0d3a4335c8/packages/block-editor/src/components/iframe/index.js#L258), I think we need a fallback background color for when the theme doesn't have a background color.

Otherwise, the site canvas will be black. Below is a screenshot of the Emptyhybrid theme enabled on `http://localhost:8889/`

![image](https://github.com/user-attachments/assets/0dc8ed75-ac02-46b4-beff-a870c259dec2)

Reply:
Oooh well spotted, we've been tripped up by the absence of background a few times before 😅